File: CodingSamples/DotNet/Fundamentals/Fundamentals/Inheritance/InterfaceTypeTest/Banking/CurrentAccount.cs
namespace Banking;

//a class declared with sealed keyword cannot be used as a base class
sealed class CurrentAccount : Account
{
    public override void Deposit(decimal amount)
    {
        if(Balance < 0)
            amount -= 500;
        Balance += amount;
    }

    public override void Withdraw(decimal amount)
    {
        Balance -= amount;
    }
}

File: CodingSamples/Java/Fundamentals/Inheritance/InterfaceTest/bank/banking/CurrentAccount.java
package banking;

//a class declared with final keyword can be subclassed
final class CurrentAccount extends Account {
    
    public void deposit(double amount) {
        if(balance < 0)
            amount -= 500;
        balance += amount;
    }

    public void withdraw(double amount) throws InsufficientFundsException {
        balance -= amount;
    }
}
